### ISSUE TITLE
ST_Intersects Out Of Memory error

### ISSUE THREAD

**Issue by RaczeQ** [duckdblabs/duckdb_spatial#86]
I've tried to intersect together a relatively small dataset and save the results to the table (or even get it as a DataFrame in Python), but no matter what memory limit I set on the database, it always ends in an OOM exception. I'm suspecting some memory leak, but can't debug it myself.

Here is the code to reproduce (python client) with files:
[regions.parquet](https://drive.google.com/file/d/1ee3a5hHsCg0WG3TZIeE068R0EmeremPN/view?usp=sharing)
[features.parquet](https://drive.google.com/file/d/1Y87psi_Cm3L-O4dml5VvmXG6MPJ8qIae/view?usp=sharing)
```python
import duckdb

CONNECTION = duckdb.connect(
    database=":memory:",
    config=dict(
        temp_directory="duckdb_temp/",
        memory_limit="2GB",
    ),
)
for extension in ['spatial', 'parquet']:
    CONNECTION.install_extension(extension)
    CONNECTION.load_extension(extension)

regions = CONNECTION.read_parquet('regions.parquet').query('r', 'SELECT * EXCLUDE(wkt), ST_GeomFromText(wkt) geometry FROM r')
features = CONNECTION.read_parquet('features.parquet').query('f', 'SELECT * EXCLUDE(wkt), ST_GeomFromText(wkt) geometry FROM f')

filled_query = """
CREATE TABLE joined_features AS
SELECT
    regions.region_id,
    features.feature_id
FROM ({regions_query}) regions
JOIN ({features_query}) features
ON ST_Intersects(regions.geometry, features.geometry)
""".format(regions_query = regions.sql_query(), features_query = features.sql_query())

print(filled_query)

# CREATE TABLE joined_features AS
# SELECT
#     regions.region_id,
#     features.feature_id
# FROM (SELECT * EXCLUDE (wkt), st_geomfromtext(wkt) AS geometry FROM r) regions
# JOIN (SELECT * EXCLUDE (wkt), st_geomfromtext(wkt) AS geometry FROM f) features
# ON ST_Intersects(regions.geometry, features.geometry)

CONNECTION.execute(filled_query)
```

**Comment by Maxxen** [comment]
Hi! Thanks for bringing this to my attention, I'm looking into it.

**Comment by Maxxen** [comment]
I've located the issue, it's a memory leak in our geometry->geos conversion. I'll push a fix in a minute.

**Comment by Maxxen** [comment]
Fix in #87 

Thanks again for filing this issue. I'm going to cut a new release later this week so you can get this fix and all the other recent changes through the CLI on nightly duckdb builds.

**Comment by RaczeQ** [comment]
That's great! Thank you very much for such responsiveness 😎
